clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/NodeStateChangeChecker.java

```diff
@@ -107,6 +110,45 @@ private static boolean noChanges(NodeState oldWantedState, NodeState newWantedSt
                && Objects.equals(newWantedState.getDescription(), oldWantedState.getDescription());
     }
 
+    private static Optional<Result> checkZeroEntriesStoredOnContentNode(HostInfo hostInfo, int nodeIndex) {
+        var entriesMetric = hostInfo.getMetrics().getValueAt(ENTRIES_METRIC_NAME, DEFAULT_SPACE_METRIC_DIMENSIONS);
+        if (entriesMetric.isEmpty() || entriesMetric.get().getLast() == null) {
+            // To allow for rolling upgrades in clusters with content node versions that do not report
+            // an entry count, defer to legacy bucket count check if the entry metric can't be found.
+            return Optional.empty();
+        }
+        long lastEntries = entriesMetric.get().getLast();
+        if (lastEntries > 0) {
+            return Optional.of(disallow("The storage node stores %d document entries".formatted(lastEntries)));
```

Comment:
It may be interesting to report back the breakdown of docs vs tombstones in this message?
An even more verbose message would also print the legacy number of buckets so it is possible to see when there is an inconsistency there.

Reply:
Will change the returned message to be of the format `The storage node stores X documents and Y tombstones across Z buckets`